src/clients/identity/types.ts

```diff
@@ -0,0 +1,92 @@
+export enum AuthHeaderType {
+    JWT = 'JWT',
+    AccessToken = 'AccessToken'
+}
+
+export interface AuthHeader {
+    token: string;
+    type: AuthHeaderType;
+}
+
+export interface IWithAuthenticationOptions {
+    roles?: string[];
+    permissions?: string[];
+}
+
+export enum tokenTypes {
+    UserApiToken = 'userApiToken',
+    TenantApiToken = 'tenantApiToken',
+    UserToken = 'userToken',
+    TenantAccessToken = 'tenantAccessToken',
+    UserAccessToken = 'userAccessToken',
+}
+
+export type Role = string;
+export type Permission = string;
+
+export type TEntity =
+    IUser |
+    IUserApiToken |
+    ITenantApiToken |
+    IUserAccessToken |
+    IUserAccessToken & IEntityWithRoles |
```

Comment:
I think it might be better to create another interface - IUserAccessTokenWithRoles

Reply:
done